Assets/Scripts/Shop/InventoryUIController.cs

```diff
@@ -0,0 +1,259 @@
+            RepositionItems();
+        }
+        else
+        {
+            Debug.LogWarning($"Tried to remove {itemType}, but it wasn't in the orderedItems list.");
+        }
+    }
+
+    private void RepositionItems()
+    {
+        for (int i = 0; i < orderedItems.Count; i++)
+        {
+            if (filledSlots[i] != null)
+            {
+                RectTransform itemRectTransform = filledSlots[i].GetComponent<RectTransform>();
+                float inventoryItemWidth = 35f;
+                itemRectTransform.anchoredPosition = new Vector2(i * inventoryItemWidth, 0);
+            }
+        }
+
+        // Move empty slots to the end
+        for (int i = orderedItems.Count; i < emptySlots.Length; i++)
+        {
+            if (emptySlots[i] != null)
+            {
+                RectTransform emptySlotRectTransform = emptySlots[i].GetComponent<RectTransform>();
+                float inventoryItemWidth = 35f;
+                emptySlotRectTransform.anchoredPosition = new Vector2(i * inventoryItemWidth, 0);
+                emptySlots[i].SetActive(true);
+            }
+        }
+    }
+
+    private void CreateInventoryItemUI(Item.ItemType itemType, int count, int index)
+    {
+        Transform inventoryItemTransform = Instantiate(inventoryItemTemplate, container);
+        RectTransform inventoryItemRectTransform = inventoryItemTransform.GetComponent<RectTransform>();
+
+        float inventoryItemWidth = 35f;
+        inventoryItemRectTransform.anchoredPosition = new Vector2(
+            index * inventoryItemWidth,
+            0
+        );
+
+        Text itemName = inventoryItemTransform.Find("ItemName").GetComponent<Text>();
+        Text itemQuantity = inventoryItemTransform.Find("ItemQuantity").GetComponent<Text>();
+
+        itemName.text = Item.GetName(itemType);
+        itemQuantity.text = 'x' + count.ToString();
+
+        inventoryItemTransform.gameObject.SetActive(true);
+
+        inventoryItems[itemType] = inventoryItemTransform.gameObject;
+
+        filledSlots[index] = inventoryItemTransform.gameObject;
+        emptySlots[index].SetActive(false);
+    }
+
+    private GameObject CreateEmptyInventoryItemUI(int index)
+    {
+        Transform inventoryItemTransform = Instantiate(emptyInventoryItemTemplate, container);
+        RectTransform inventoryItemRectTransform = inventoryItemTransform.GetComponent<RectTransform>();
+
+        float inventoryItemWidth = 35f;
+        inventoryItemRectTransform.anchoredPosition = new Vector2(
+            index * inventoryItemWidth,
+            0
+        );
+
+        inventoryItemTransform.gameObject.SetActive(true);
+
+        return inventoryItemTransform.gameObject;
+    }
+
+    private void ActivateItem(Item.ItemType itemType)
+    {
+        Debug.Log($"Activated item: {Item.GetName(itemType)}");
+
+        if (itemType == activeItem)
+        {
+            PlayerWeaponManager.SwitchToPistol();
+            activeItem = Item.ItemType.Subway8;
```

Comment:
uhhh lol

Reply:
lol. this was a forced beta hack. i hate it too.

basically, this if statement is run if you toggle off the active inventory item (i.e. you have the rocket launcher enabled, and you press its button again to put it away). I need the activeItem to no longer be the rocket launcher so that you can reopen it, so I set it to the footlong lolllll

Post beta I will just make a new ItemType for the pistol, but i have to make it not show up in the shop menu and other stuff, so let's get this out rq then fix that